### PR TITLE
Fix search results overwriting root limit

### PR DESCRIPTION
**Pull request by aschempp**
Fixes https://github.com/contao/contao/issues/7366

Be aware that the only actual fix is removing `eliminateNestedPages` from the search results. The check makes no sense for search results. Since we already manually merge the results with `$this->root`, the permissions are applied as well (manually checked this).

The other changes are
- better naming (`$objRoot` to `$objFound` because these are search results, merged into `$arrFound` later)
- Removal of the check for `sorting` field in the table, since a tree must always have a sorting field (that's already enforced on line 3672 with `Table can not be shown as tree, because the "id", "pid" or "sorting" field is missing!`)
- Therefore removal of the third `if/else` condition since we only support the two known DC modes (5 + 6)
- Renamed the mode 6 result from `SELECT pid` to `SELECT pid AS id` so the retrieval field `$objFound->id` is the same for both modes.

### CODE REVIEW

**Review by zoglo — DISMISSED** [review]
Did test this out and that one looks good.

I opened a PR to refactor it further and just check for the extended tree mode once in this method:
https://github.com/aschempp/contao/pull/12

**Review by zoglo — DISMISSED** [review]

File: core-bundle/src/Resources/contao/drivers/DC_Table.php [hunk 1]
```diff
@@ -3795,7 +3786,7 @@ protected function treeView()
 		{
 			for ($i=0, $c=\count($topMostRootIds); $i<$c; $i++)
 			{
-				$tree .= $this->generateTree($table, $topMostRootIds[$i], array('p'=>($topMostRootIds[($i-1)] ?? null), 'n'=>($topMostRootIds[($i+1)] ?? null)), $blnHasSorting, -20, ($blnClipboard ? $arrClipboard : false), (($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['mode'] ?? null) == self::MODE_TREE && $blnClipboard && $topMostRootIds[$i] == $arrClipboard['id']), false, false, $arrFound);
+				$tree .= $this->generateTree($table, $topMostRootIds[$i], array('p'=>($topMostRootIds[($i-1)] ?? null), 'n'=>($topMostRootIds[($i+1)] ?? null)), true, -20, ($blnClipboard ? $arrClipboard : false), (($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['mode'] ?? null) == self::MODE_TREE && $blnClipboard && $topMostRootIds[$i] == $arrClipboard['id']), false, false, $arrFound);
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/drivers/DC_Table.php]: Why did you remove `$blnHasSorting` in line 3725 and set it to `true` here? Not every tree view table has a `sorting` column.

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/drivers/DC_Table.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php]: yes it has. It is enforced in https://github.com/contao/contao/blob/4.13/core-bundle/src/Resources/contao/drivers/DC_Table.php#L3669-L3673

Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php]: I looks like however we could have an _extended tree_ without sorting? Has anyone ever tried or used that?

Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:3789]: As discussed on the Contao Call yesterday I reverted the `$blnHasSorting` removal in 34c245813989e7a117ebbc15543402d84768c9e4

Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php]: As discussed on the Contao Call yesterday I reverted the `$blnHasSorting` removal in 34c245813989e7a117ebbc15543402d84768c9e4

File: core-bundle/src/Resources/contao/drivers/DC_Table.php [hunk 2]
```diff
@@ -3727,48 +3729,43 @@ protected function treeView()
...
-				$objRoot = $this->Database->prepare("SELECT pid, (SELECT sorting FROM " . $table . " WHERE " . $this->strTable . ".pid=" . $table . ".id) AS psort FROM " . $this->strTable . " WHERE " . implode(' AND ', $this->procedure) . " GROUP BY pid ORDER BY psort, pid")
-										  ->execute($this->values);
+				$objFound = $this->Database->prepare("SELECT pid AS id, (SELECT sorting FROM " . $table . " WHERE " . $this->strTable . ".pid=" . $table . ".id) AS psort FROM " . $this->strTable . " WHERE " . implode(' AND ', $this->procedure) . " GROUP BY pid ORDER BY psort, pid")
+					->execute($this->values);
 			}
 			else
 			{
-				$objRoot = $this->Database->prepare("SELECT pid FROM " . $this->strTable . " WHERE " . implode(' AND ', $this->procedure) . " GROUP BY pid")
-										  ->execute($this->values);
+				$objFound = $this->Database->prepare("SELECT pid AS id FROM " . $this->strTable . " WHERE " . implode(' AND ', $this->procedure) . " GROUP BY pid")
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/drivers/DC_Table.php:3744]: Can someone tell me if we need to use `GROUP BY pid` or `GROUP BY id` due to the field being aliased?

Comment (zoglo) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:3744]: Order of execution, the alias is set within the SELECT so it's not replaced in the group by.
https://sqlbolt.com/lesson/select_queries_order_of_execution

1. FROM/JOIN
2. WHERE
3. GROUP BY
4. HAVING
5. SELECT
6. DISTINCT
7. ORDER BY
8. LIMIT/OFFSET

This is different to the order of clause (when writing the query) so `GROUP BY pid` is right in your context.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.
